### PR TITLE
s3 error handling

### PR DESCRIPTION
**Pull request by hugoesthere**
Error handling for creation & listing of s3 buckets.

@kevgo @trushton @martinjaime

### CODE REVIEW

**Comment by hugoesthere** [comment]
@kevgo could you take a look at this?

File: exo-deploy/src/deployers/aws-deployer.ls [hunk 1]
```diff
@@ -59,14 +59,15 @@ class AwsDeployer
   _create-remote-store: (done) ->
     @s3
       ..create-bucket Bucket: @aws-config['remote-state-store'], CreateBucketConfiguration: LocationConstraint: "#{@aws-config.region}", (err, data) ~>
-          if err then return done new Error err #TODO: inject logger object
+          | err => return done process.stdout.write err.message
```
Comment (kevgo) [inline exo-deploy/src/deployers/aws-deployer.ls]:
```ls
| err  =>  logger.log "Cannot create S3 bucket: #{err.message}" ; return done err

File: exo-deploy/src/deployers/aws-deployer.ls [hunk 2]
```diff
@@ -59,14 +59,15 @@ class AwsDeployer
...
           ..put-bucket-versioning Bucket: @aws-config['remote-state-store'], VersioningConfiguration: {Status: 'Enabled'}, ~>
             ..put-object Bucket: @aws-config['remote-state-store'], Key: 'terraform.tfstate', done
 
 
   # verify that s3 bucket with bucket-name exists
   _has-bucket: (bucket-name, done) ->
     @s3.list-buckets (err, data) ~>
+      | err => return process.stdout.write err.message
```
Comment (kevgo) [inline exo-deploy/src/deployers/aws-deployer.ls]: Same here

**Comment by hugoesthere** [comment]
@kevgo comments addressed

**Review by kevgo — APPROVED** [review]

**Pull request closed without merging**